src/parsing/convertXmlToAst.ts

```diff
@@ -14,7 +15,7 @@ export default function convertXmlToAst(element: Element): IAST {
 		element.namespaceURI !== BUILT_IN_NAMESPACE_URIS.FONTOXPATH_NAMESPACE_URI &&
 		element.namespaceURI !== BUILT_IN_NAMESPACE_URIS.XQUERYX_UPDATING_NAMESPACE_URI
 	) {
-		throw new Error('AST should only contain XQUERYX elements');
+		throw errXPTY0004('The XML structure passed as an XQueryX program was not valid XQueryX');
```

Comment:
Not entirely sure about this one, the fontoxpath:evaluate test expected XPTY0004, and it seems that there were no other tests expecting anything else if an incorrect XQueryX AST is passed directly to evaluateXPath...

Reply:
Sounds like a proper error code: some type is unexpected